src/monodroid/jni/timing.hh

```diff
@@ -124,7 +136,7 @@ namespace xamarin::android
 
 			std::lock_guard<std::mutex> lock (sequence_lock);
 			if (sequence->dynamic) {
-				memset (sequence, 0, sizeof(*sequence));
+				sequence->period.reset ();
```

Comment:
This is odd...and I don't remember this coming up before... but why clear out `period` if it'll be invalidated on the next `delete sequence` line?  Why is this done?  (Same question applies to the original `memset()`, and I don't remember this being discussed previously with PR #3729.)

In principal, the `delete sequence` could (should?) "scribble" over `*sequence` so that if it's reused, it can be more readily detected in a debugger.  We'd thus effectively have:

```c
memset (sequence, 0, sizeof (*sequence);
memset (sequence, 0xbb, sizeof (*sequence);  // or some other "`free` was here!" constant
```

What's the `memset()`/`period.reset()` buy us?

Reply:
The reason clearing is done is that we don't control the `IntPtr` used to hold the sequence on the managed side. And clearing the sequence to 0 marks it as (sort of) "unused memory" should anything in the managed land reuse the pointer (which hopefully never happens). Doing the `memset` in our implementation of `delete` would unnecessarily slow things down.

Reply:
> Doing the `memset` in our implementation of `delete` would unnecessarily slow things down.

I wasn't advocating that *our* `delete` do that.  I was just saying that *many* `free()`/`operator delete()` implementations do so.